File: Kotlin/27.remove_element.kt
fun main() {
    println(removeElement(intArrayOf(3,2,2,3), 3))
    println(removeElement(intArrayOf(0,1,2,2,3,0,4,2), 2))
}

fun removeElement(nums: IntArray, `val`: Int): Int {
    val list = mutableListOf<Int>()

    nums.forEach { e -> 
        if(e!=`val`) {
            list.add(e)
        }
    }

    list.forEachIndexed { index, value ->
        nums[index] = value
    }

    return list.size;
}

File: Dart/27.remove_element.dart
  void main() {
    print(removeElement([3,2,2,3], 3));
    print(removeElement([0,1,2,2,3,0,4,2], 2));
  }
  
  int removeElement(List<int> nums, int val) {
    for(int i = nums.length - 1; i >=0 ; i--) {
      if(nums[i]== val) {
        nums.removeAt(i);
      }
    }

    return nums.length;
  }